rmb-sdk-go/peer/connection.go

```diff
@@ -157,23 +161,49 @@ func (c *InnerConnection) connect() (*websocket.Conn, error) {
 		return nil, errors.Wrap(err, "could not create new jwt")
 	}
 
-	relayURL := fmt.Sprintf("%s?%s", c.url, token)
+	var (
+		con        *websocket.Conn
+		resp       *http.Response
+		maxRetries uint64 = 2
+		i                 = rand.Intn(len(c.urls))
+	)
 
-	con, resp, err := websocket.DefaultDialer.Dial(relayURL, nil)
-	if err != nil {
-		var body []byte
-		var status string
-		if resp != nil {
-			status = resp.Status
-			body, _ = io.ReadAll(resp.Body)
+	// implementation is not thread-safe.
+	c.m.Lock()
+	defer c.m.Unlock()
+
+	b := backoff.WithMaxRetries(
+		backoff.NewConstantBackOff(time.Nanosecond),
+		maxRetries*uint64(len(c.urls)),
+	)
+
+	err = backoff.RetryNotify(func() error {
+		url := c.urls[i%len(c.urls)]
+		relayURL := fmt.Sprintf("%s?%s", url, token)
+		log.Debug().Str("url", url).Msg("connecting")
+
+		con, resp, err = websocket.DefaultDialer.Dial(relayURL, nil)
+		if err != nil {
+			var body []byte
+			var status string
+			if resp != nil {
+				status = resp.Status
+				body, _ = io.ReadAll(resp.Body)
+			}
+
+			i++
+			return errors.Wrapf(err, "failed to connect (%s): %s", status, string(body))
 		}
 
-		return nil, errors.Wrapf(err, "failed to connect (%s): %s", status, string(body))
-	}
+		if resp.StatusCode != http.StatusSwitchingProtocols {
+			i++
+			return fmt.Errorf("invalid response %s", resp.Status)
+		}
 
-	if resp.StatusCode != http.StatusSwitchingProtocols {
-		return nil, fmt.Errorf("invalid response %s", resp.Status)
-	}
+		return nil
+	}, b, func(err error, _ time.Duration) {
+		log.Error().Err(err).Msg("failed to connect to relay, retrying")
+	})
 
-	return con, nil
+	return con, err
```

Comment:
This will open a connection to a single relay and return with the first successful attempt. This approach will create issues when delivering messages to the client. 
Imagine having two email addresses that your friends can use any of them to message you, but you only check one of them.

Instead, you should enable the client to open multiple connections to relays simultaneously, And constantly listen for anything passed over any of these connections. If a connection to one of the relays fails to establish, the client should asynchronously keep trying as long as it is running. And if a connection to one of the relays drops, the client should attempt to reconnect.